### PR TITLE
feat: add status and headers overwrite for HandlerContext.render

### PR DESCRIPTION
**Pull request by crowlKats**

### CODE REVIEW

File: src/server/types.ts [hunk 1]
```diff
@@ -71,10 +71,17 @@ export interface RouteConfig {
   csp?: boolean;
 }
 
+export interface RenderOptions {
+  statusOverwrite?: number;
```
Comment (lucacasonato) [inline src/server/types.ts]:
```suggestion
  status?: number;
  statusText?: string;
```

Let's add this option too while we're at it.

**Comment by jiawei397** [comment]
It seems that I need this merger. Is there any plan?

**Comment by iuioiua** [comment]
Hey Leo, could you provide some uses for this feature, and is this ready for review?

**Comment by crowlKats** [comment]
Yes its ready for review.
Example usecases:
https://github.com/denoland/dotland/issues/2256
https://github.com/denoland/dotland/issues/2258

**Review by lucacasonato — APPROVED** [review]
LGTM

**Pull request merged**